js/models/messages.js

```diff
@@ -1935,105 +1977,212 @@
 
-              if (!messageAllowed) {
-                confirm();
-                return null;
-              }
+            // Returning true drops the message
+            if (!messageAllowed) {
+              confirm();
+              return true;
             }
           }
-          // For every member, see if we need to establish a session:
-          initialMessage.group.members.forEach(memberPubKey => {
-            const haveSession = _.some(
-              textsecure.storage.protocol.sessions,
-              s => s.number === memberPubKey
-            );
+        }
+        // send a session request for all the members we do not have a session with
+        window.libloki.api.sendSessionRequestsToMembers(
+          initialMessage.group.members
+        );
+      } else if (newGroup) {
+        // We have an unknown group, we should request info from the sender
+        textsecure.messaging.requestGroupInfo(conversationId, [primarySource]);
+      }
+      return false;
+    },
 
-            const ourPubKey = textsecure.storage.user.getNumber();
-            if (!haveSession && memberPubKey !== ourPubKey) {
-              ConversationController.getOrCreateAndWait(
-                memberPubKey,
-                'private'
-              ).then(() => {
-                textsecure.messaging.sendMessageToNumber(
-                  memberPubKey,
-                  '(If you see this message, you must be using an out-of-date client)',
-                  [],
-                  undefined,
-                  [],
-                  Date.now(),
-                  undefined,
-                  undefined,
-                  { messageType: 'friend-request', sessionRequest: true }
-                );
-              });
-            }
-          });
-        } else if (newGroup) {
-          // We have an unknown group, we should request info from the sender
-          textsecure.messaging.requestGroupInfo(conversationId, [
-            primarySource,
-          ]);
+    async handleSessionRequest(source, primarySource, confirm) {
+      // Check if the contact is a member in one of our private groups:
+      const isKnownClosedGroupMember = window
+        .getConversations()
+        .models.filter(c => c.get('members'))
+        .reduce((acc, x) => window.Lodash.concat(acc, x.get('members')), [])
+        .includes(primarySource);
+
+      libloki.api.debug.logSessionRequest(
+        `Received SESSION_REQUEST from source: ${source}, primarySource: ${primarySource}, is one of our private groups: ${isKnownClosedGroupMember}`
+      );
+
+      if (isKnownClosedGroupMember) {
+        window.log.info(
+          `Auto accepting a 'group' session request for a known group member: ${primarySource}`
+        );
+        window.libloki.api.sendBackgroundMessage(
+          source,
+          window.textsecure.OutgoingMessage.DebugMessageType
+            .SESSION_REQUEST_ACCEPT
+        );
+
+        confirm();
```

Comment:
the return is made in `message.js` in the `if (message.isSessionRequest())` case